### PR TITLE
Update documentation links

### PR DESCRIPTION
**Pull request by daverodgman**
Signed-off-by: Dave Rodgman <dave.rodgman@arm.com>

See also: https://github.com/Mbed-TLS/mbedtls-docs/pull/55
Backport: #6414 
Changelog: not needed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,9 @@ We provide some non-standard configurations focused on specific use cases in the
 Documentation
 -------------
 
-Documentation for the Mbed TLS interfaces in the default library configuration is available as part of the [Mbed TLS documentation](https://tls.mbed.org/api/).
+The main Mbed TLS documentation is available via [ReadTheDocs](https://mbed-tls.readthedocs.io/).
+
+Documentation for the PSA Crypto interface is available [on GitHub](https://armmbed.github.io/mbed-crypto/psa/#application-programming-interface).
```
Comment (daverodgman) [inline README.md:22]:
```suggestion
Documentation for the PSA Cryptography API is available [on GitHub](https://armmbed.github.io/mbed-crypto/psa/#application-programming-interface).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ As mentioned, tests that show the correctness of the feature or bug fix should b
 
 Mbed TLS includes a comprehensive set of test suites in the `tests/` directory that are dynamically generated to produce the actual test source files (e.g. `test_suite_mpi.c`). These files are generated from a `function file` (e.g. `suites/test_suite_mpi.function`) and a `data file` (e.g. `suites/test_suite_mpi.data`). The function file contains the test functions. The data file contains the test cases, specified as parameters that will be passed to the test function.
 
-[A Knowledge Base article describing how to add additional tests is available on the Mbed TLS website](https://tls.mbed.org/kb/development/test_suites).
+[A Knowledge Base article describing how to add additional tests is available on the Mbed TLS website](https://mbed-tls.readthedocs.io/en/latest/kb/development/test_suites/).
 
 A test script `tests/scripts/basic-build-test.sh` is available to show test coverage of the library. New code contributions should provide a similar level of code coverage to that which already exists for the library.
```
Comment (tom-cosgrove-arm) [inline CONTRIBUTING.md:61]: This could link to `https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/basic-build-test.sh`

File: SUPPORT.md [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 Here are some useful sources of information about using Mbed TLS:
 
+- [ReadTheDocs](https://mbed-tls.readthedocs.io/);
 - API documentation, see the [Documentation section of the
   README](README.md#License);
 - the `docs` directory in the source tree;
-- the [Mbed TLS knowledge Base](https://tls.mbed.org/kb);
```
Comment (tom-cosgrove-arm) [inline SUPPORT.md:8]: why remove this rather than change the link to `https://mbed-tls.readthedocs.io/en/latest/kb/`?

File: SUPPORT.md [hunk 4]
```diff
@@ -2,10 +2,10 @@
 
 Here are some useful sources of information about using Mbed TLS:
 
+- [ReadTheDocs](https://mbed-tls.readthedocs.io/);
 - API documentation, see the [Documentation section of the
   README](README.md#License);
```
Comment (paul-elliott-arm) [inline SUPPORT.md:7]:
```suggestion
  README](README.md#documentation);
```
Not part of this commit, but why is this pointing to the Licence section?

File: docs/3.0-migration-guide.md [hunk 5]
```diff
@@ -141,7 +141,7 @@ recommended), or users who used it through the entropy module but had it as the
 only source of entropy. If you're in that case, please declare OS or hardware
 RNG interfaces with `mbedtls_entropy_add_source()` and/or use an entropy seed
 file created securely during device provisioning. See
-<https://tls.mbed.org/kb/how-to/add-entropy-sources-to-entropy-pool> for more
+<https://mbed-tls.readthedocs.io/en/latest/kb/how-to/add-entropy-sources-to-entropy-pool> for more
```
Comment (tom-cosgrove-arm) [inline docs/3.0-migration-guide.md:144]: Should this link have a trailing `/` like the others?

File: docs/architecture/alternative-implementations.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ The general principle of an alternative implementation is:
 * Create a header file `xxx_alt.h` that defines the context type(s) used by the module. For example, `mbedtls_aes_context` for AES.
 * Implement all the functions from the module, i.e. the functions declared in `include/mbedtls/xxx.h`.
 
-See https://tls.mbed.org/kb/development/hw_acc_guidelines for a more detailed guide.
+See https://mbed-tls.readthedocs.io/en/latest/kb/development/hw_acc_guidelines for a more detailed guide.
```
Comment (tom-cosgrove-arm) [inline docs/architecture/alternative-implementations.md:41]: Should this link have a trailing `/` like the others?

File: docs/architecture/testing/test-framework.md [hunk 7]
```diff
@@ -6,7 +6,7 @@ This document is incomplete. You can help by expanding it.
 
 ## Unit tests
 
-See <https://tls.mbed.org/kb/development/test_suites>
+See <https://mbed-tls.readthedocs.io/en/latest/kb/development/test_suites>
```
Comment (tom-cosgrove-arm) [inline docs/architecture/testing/test-framework.md:9]: Trailing `/`

File: SUPPORT.md (hunk 3 above)
Comment (daverodgman) [reply SUPPORT.md:8]: It's a subset of the ReadTheDocs link added at the top - seems unnecessarily redundant?

File: docs/architecture/tls13-support.md [hunk 8]
```diff
@@ -181,7 +181,7 @@ Coding rules checklist for TLS 1.3
 The following coding rules are aimed to be a checklist for TLS 1.3 upstreaming
 work to reduce review rounds and the number of comments in each round. They
 come along (do NOT replace) the project coding rules
-(https://tls.mbed.org/kb/development/mbedtls-coding-standards). They have been
+(https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards). They have been
```
Comment (tom-cosgrove-arm) [inline docs/architecture/tls13-support.md:184]: Trailing `/`

File: docs/3.0-migration-guide.md (hunk 5 above)
Comment (daverodgman) [reply docs/3.0-migration-guide.md:144]: The links work correctly either way. Agreed, it would be nicer if I'd been consistent but not sure it's worth fixing?

File: include/mbedtls/mbedtls_config.h [hunk 9]
```diff
@@ -2552,7 +2552,7 @@
  *
  * \note See also our Knowledge Base article about porting to a new
  * environment:
- * https://tls.mbed.org/kb/how-to/how-do-i-port-mbed-tls-to-a-new-environment-OS
+ * https://mbed-tls.readthedocs.io/en/latest/kb/how-to/how-do-i-port-mbed-tls-to-a-new-environment-OS
```
Comment (tom-cosgrove-arm) [inline include/mbedtls/mbedtls_config.h:2555]: No trailing `/` on any of these (but they've been added in several of the changes further up this PR)

**Review by paul-elliott-arm — CHANGES_REQUESTED** [review]

File: include/mbedtls/ssl.h [hunk 10]
```diff
@@ -2322,7 +2322,7 @@ int mbedtls_ssl_check_record( mbedtls_ssl_context const *ssl,
  *                 here, except if using an event-driven style.
  *
  * \note           See also the "DTLS tutorial" article in our knowledge base.
- *                 https://tls.mbed.org/kb/how-to/dtls-tutorial
+ *                 https://mbed-tls.readthedocs.io/en/latest/kb/how-to/dtls-tutorial
```
Comment (tom-cosgrove-arm) [inline include/mbedtls/ssl.h:2325]: Trailing `/`

File: CONTRIBUTING.md (hunk 2 above)
Comment (daverodgman) [reply CONTRIBUTING.md:61]: There are quite a few references to files in the various .md documents - the others don't link into Github, so this is consistent.

File: SUPPORT.md (hunk 3 above)
Comment (tom-cosgrove-arm) [reply SUPPORT.md:8]: It seems like a useful subset to specifically point out (which we do in a number of places)

File: docs/3.0-migration-guide.md (hunk 5 above)
Comment (tom-cosgrove-arm) [reply docs/3.0-migration-guide.md:144]: > it would be nicer if I'd been consistent but not sure it's worth fixing?

You're the PR author - that decision is yours. It's not a blocker, just inconsistent, so I've pointed it out

File: SUPPORT.md [hunk 11]
```diff
@@ -2,10 +2,10 @@
...
 - the `docs` directory in the source tree;
-- the [Mbed TLS knowledge Base](https://tls.mbed.org/kb);
 - the [Mbed TLS mailing-list
```
Comment (daverodgman) [inline SUPPORT.md:9]:
```suggestion
- the [Mbed TLS knowledge Base](https://mbed-tls.readthedocs.io/en/latest/kb/);
- the [Mbed TLS mailing-list
```

File: SUPPORT.md (hunk 3 above)
Comment (daverodgman) [reply SUPPORT.md:8]: OK, restored

File: SUPPORT.md [hunk 12]
```diff
@@ -2,10 +2,11 @@
 
 Here are some useful sources of information about using Mbed TLS:
 
+- [ReadTheDocs](https://mbed-tls.readthedocs.io/);
 - API documentation, see the [Documentation section of the
-  README](README.md#License);
+  README](README.md#documentation);
 - the `docs` directory in the source tree;
-- the [Mbed TLS knowledge Base](https://tls.mbed.org/kb);
+- the [Mbed TLS knowledge Base](https://mbed-tls.readthedocs.io/en/latest/kb/);
```
Comment (daverodgman) [inline SUPPORT.md:9]:
```suggestion
- the [Mbed TLS Knowledge Base](https://mbed-tls.readthedocs.io/en/latest/kb/);
```

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]

**Review by paul-elliott-arm — APPROVED** [review]
LGTM
